crates/rpc/types/src/types.rs

```diff
@@ -363,6 +364,21 @@ impl From<CheckpointEntry> for RpcCheckpointInfo {
     }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+/// Deposit entry for RPC corresponding to [`WithdrawalIntent`]
+pub struct RpcWithdrawalIntent {
+    /// Corresponding deposit id
+    pub deposit_idx: u32,
+    /// Corresponding deposit txid
+    pub deposit_txid: Txid,
+    /// Quantity of L1 asset, for Bitcoin this is sats.
+    pub amt: BitcoinAmount,
+    /// Destination [`Descriptor`] for the withdrawal
+    pub destination: Descriptor,
+    /// operator index
+    pub operator_idx: OperatorIdx,
+}
```

Comment:
This should be `RpcWithdrawalAssignment`.  Intents are basically just messages with the destination and amount, when they're assigned to a deposit and operator then that produces an assignment.